contracts/bonding/BondingManager.sol

```diff
@@ -505,15 +504,17 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             decreaseTotalStake(currentDelegate, del.bondedAmount, _oldDelegateNewPosPrev, _oldDelegateNewPosNext);
         }
 
-        Transcoder storage newDelegate = transcoders[_to];
-        EarningsPool.Data storage currPool = newDelegate.earningsPoolPerRound[currentRound];
-        if (currPool.cumulativeRewardFactor == 0) {
-            currPool.cumulativeRewardFactor = cumulativeFactorsPool(newDelegate, newDelegate.lastRewardRound)
-                .cumulativeRewardFactor;
-        }
-        if (currPool.cumulativeFeeFactor == 0) {
-            currPool.cumulativeFeeFactor = cumulativeFactorsPool(newDelegate, newDelegate.lastFeeRound)
-                .cumulativeFeeFactor;
+        {
```

Comment:
Do we need these brackets around L508-517?

Reply:
Added to avoid "Stack too deep" error by reducing the local variables in the function scope.